Fix http user catting wip

mv2731 opened the pull request:
Discovered a small bug after many rounds of testing and reading the log. Because we are using dstr_catf to copy the libobs and OBS version to http_user_agent, it was catting this string onto muxer_settings every single time the hls_start_function runs. I added a check to ensure that this is not set unless muxer_settings is already empty.

mv2731 committed:
obs-ffmpeg: Check muxer_settings not set

Before cat-ing the user agent to muxer_settings, check if
it has already been set, and if it has, do not set it.

mv2731 committed:
obs-ffmpeg: Fix errors in added lines

ushadow commented on plugins/obs-ffmpeg/obs-ffmpeg-mux.c:
```diff
@@ -388,8 +388,9 @@ static bool ffmpeg_hls_mux_start(void *data)
-	dstr_catf(&stream->muxer_settings, "http_user_agent=libobs/%s",
-		  OBS_VERSION);
+	if (dstr_is_empty(&stream->muxer_settings))
+		dstr_catf(&stream->muxer_settings, "http_user_agent=libobs/%s",
```
I think it's better to use dstr_init to reinit muxer_settings and then concatenate the string. Then it's not necessary to check if it's empty or not.

ushadow reviewed (changes requested):
Nice find!

mv2731 committed:
obs-ffmpeg: Add dstr_init to zero out muxer_settings

Add a call to dstr_init in the hls_start_mux function so that
every time hls_start_mux is called muxer_settings is zeroed
and can be set again.
